### PR TITLE
Option for klipper only: Adjustment of "accel_to_decel"

### PR DESCRIPTION
**Pull request by cotepat**
Add option to automatically adjust accel_to_decel to 50% of chosen acceleration. This option is only displayed when using klipper.

image

### CODE REVIEW

**Comment by cotepat** [comment]
What would you think of this? I still think that most people using klipper would want/expect this behavior... But will not write a macro. Should we put the value (50%) in a specific field also?

File: src/libslic3r/Print.cpp [hunk 1]
```diff
@@ -103,7 +103,6 @@ bool Print::invalidate_state_by_config_options(const ConfigOptionResolver & /* n
         "outer_wall_acceleration",
         "inner_wall_acceleration",
         "initial_layer_acceleration",
-        "outer_wall_acceleration",
```
Comment (cotepat) [inline src/libslic3r/Print.cpp:106]: this was a duplicated entry... did some cleanup

File: src/slic3r/GUI/Tab.cpp [hunk 2]
```diff
@@ -3633,7 +3634,7 @@ void TabPrinter::toggle_options()
     }
 
     auto gcf = m_config->option<ConfigOptionEnum<GCodeFlavor>>("gcode_flavor")->value;
-        if (m_active_page->title() == "Motion ability") {
```
Comment (cotepat) [inline src/slic3r/GUI/Tab.cpp:3636]: indentation was incorrect. fixed it

**Comment by The-Futur1st** [comment]
It is not a rule to have 50% of acceleration as accel_to_decel. In most cases people do it because it is written in the documentation. I am prefer 75%, because my printer survive it without any issues. You definitely should to consider to make float value(0.0-1.0) rather than predefined 50%.

**Comment by cotepat** [comment]
> It is not a rule to have 50% of acceleration as accel_to_decel. In most cases people do it because it is written in the documentation as a good value. I am prefer 75%, because my printer survive it without any issues. You definitely should to consider to make float value(0.0-1.0) rather than predefined 50%.

Awesome, I will do that.

**Comment by cotepat** [comment]
Ok it is now a variable setting (range 1-100% because 0 fails on klipper...) : 
image

image

File: src/libslic3r/PrintConfig.cpp [hunk 3]
```diff
@@ -1168,7 +1168,7 @@ void PrintConfigDef::init_fff_params()
 
     def = this->add("pressure_advance", coFloats);
     def->label = L("Pressure advance");
-    def->tooltip = L("Pressure advnce(Klipper) AKA Linear advance factor(Marlin)");
```
Comment (SoftFever) [inline src/libslic3r/PrintConfig.cpp:1171]: nice catch ;)

File: src/libslic3r/GCodeWriter.cpp [hunk 4]
```diff
@@ -176,10 +176,14 @@ std::string GCodeWriter::set_acceleration(unsigned int acceleration)
...
         // Use M204 P, we don't want to override travel acc by M204 S (which is deprecated anyway).
         gcode << "M204 P" << acceleration;
-    } else {
-        // M204: Set default acceleration
+    } else if (FLAVOR_IS(gcfKlipper) && this->config.accel_to_decel_enable) {
+        gcode << "SET_VELOCITY_LIMIT ACCEL_TO_DECEL=" << acceleration * this->config.accel_to_decel_factor / 100;
+        if (GCodeWriter::full_gcode_comment) gcode << " ; adjust max_accel_to_decel to 0.5 of new accel value\n";
         gcode << "M204 S" << acceleration;
-    }
+        // Set max accel to decel to half of acceleration
```
Comment (SoftFever) [inline src/libslic3r/GCodeWriter.cpp:183]: The comment can be updated as well.

**Comment by SoftFever** [comment]
@cotepat 
Could you solve the conflict?
The PR LGTM.
We can merge it once the conflicts are solved.

**Comment by SoftFever** [comment]
Merged.
Thanks, it's a handy option 👍

**Pull request merged**